components/gitpod-protocol/src/protocol.ts

```diff
@@ -293,11 +293,24 @@ export namespace EnvVar {
             return res;
         }
 
+        // Returns a host value, which can include port, if token also has length.
+        const getHost = (parts: string[]): string => {
+            if (parts.length === 2 && parts[0] && parts[1]) {
+                return parts[0];
+            } else if (parts.length === 3 && parts[0] && parts[1] && parts[2]) {
+                return `${parts[0]}:${parts[1]}`;
+            }
+            return "";
+        };
+
         (imageAuth.value || "")
             .split(",")
             .map((e) => e.trim().split(":"))
-            .filter((e) => e.length == 2)
-            .forEach((e) => res.set(e[0], e[1]));
+            .forEach((parts) => {
+                const host = getHost(parts);
+                host !== "" ? res.set(host, parts[parts.length - 1]) : null;
```

Comment:
We are breaking the abstraction of `getHost` here, because we are not only needing the host, but the auth/password part as well.
(This surfaces in that we access part by index, without prior checking that the length >= 1 - in this function).

One way to solve is by having naming the function to `parse` and have it return `{ host: string, auth: string } | undefined`, and calling if like this:

```
const parsed = parse(parts);
if (parsed) {
    res.set(parsed.host, parsed.auth);
}
```

Reply:
Much cleaner, thank you for the suggestion! I had forgotten about this syntax: `{ host: string, auth: string } | undefined`

Reply:
Fixed in d444551a4b93038190b9d035a5c3eea3545bfb55